lightning/src/ln/channel.rs

```diff
@@ -5374,912 +4672,2113 @@ impl<Signer: WriteableEcdsaChannelSigner> Channel<Signer> {
-			cltv_expiry,
-			state: OutboundHTLCState::LocalAnnounced(Box::new(onion_routing_packet.clone())),
-			source,
-		});
+		self.context.channel_state = ChannelState::FundingSent as u32;
+		self.context.channel_id = funding_txo.to_channel_id();
+		self.context.cur_counterparty_commitment_transaction_number -= 1;
+		self.context.cur_holder_commitment_transaction_number -= 1;
 
-		let res = msgs::UpdateAddHTLC {
-			channel_id: self.channel_id,
-			htlc_id: self.next_holder_htlc_id,
-			amount_msat,
-			payment_hash,
-			cltv_expiry,
-			onion_routing_packet,
+		log_info!(logger, "Generated funding_signed for peer for channel {}", log_bytes!(self.channel_id()));
+
+		// Promote the channel to a full-fledged one now that we have updated the state and have a
+		// `ChannelMonitor`.
+		let mut channel = Channel {
+			context: self.context,
 		};
-		self.next_holder_htlc_id += 1;
+		let channel_id = channel.context.channel_id.clone();
+		let need_channel_ready = channel.check_get_channel_ready(0).is_some();
+		channel.monitor_updating_paused(false, false, need_channel_ready, Vec::new(), Vec::new(), Vec::new());
 
-		Ok(Some(res))
+		Ok((channel, msgs::FundingSigned {
+			channel_id,
+			signature,
+			#[cfg(taproot)]
+			partial_signature_with_nonce: None,
+		}, channel_monitor))
 	}
+}
 
-	fn build_commitment_no_status_check<L: Deref>(&mut self, logger: &L) -> ChannelMonitorUpdate where L::Target: Logger {
-		log_trace!(logger, "Updating HTLC state for a newly-sent commitment_signed...");
-		// We can upgrade the status of some HTLCs that are waiting on a commitment, even if we
-		// fail to generate this, we still are at least at a position where upgrading their status
-		// is acceptable.
-		for htlc in self.pending_inbound_htlcs.iter_mut() {
-			let new_state = if let &InboundHTLCState::AwaitingRemoteRevokeToAnnounce(ref forward_info) = &htlc.state {
-				Some(InboundHTLCState::AwaitingAnnouncedRemoteRevoke(forward_info.clone()))
-			} else { None };
-			if let Some(state) = new_state {
-				log_trace!(logger, " ...promoting inbound AwaitingRemoteRevokeToAnnounce {} to AwaitingAnnouncedRemoteRevoke", log_bytes!(htlc.payment_hash.0));
-				htlc.state = state;
-			}
+// A not-yet-funded outbound (from holder) channel using V2 channel establishment.
+pub(super) struct OutboundV2Channel<Signer: ChannelSigner> {
+	#[cfg(not(test))]
+	context: DualFundingChannelContext<Signer>,
+	#[cfg(test)]
+	pub context: DualFundingChannelContext<Signer>,
+}
+
+impl<Signer: WriteableEcdsaChannelSigner> OutboundV2Channel<Signer> {
+	pub fn new<ES: Deref, SP: Deref, F: Deref>(
+		fee_estimator: &LowerBoundedFeeEstimator<F>, entropy_source: &ES, signer_provider: &SP,
+		counterparty_node_id: PublicKey, their_features: &InitFeatures, channel_value_satoshis: u64,
+		funding_inputs: Vec<DualFundingUtxo>, change_script_pubkey: Script, user_id: u128,
+		config: &UserConfig, current_chain_height: u32, outbound_scid_alias: u64,
+		require_confirmed_inputs: bool,
```

Comment:
This field can be made part of the `UserConfig`. Beyond would we like to sanitize other attributes of the counterparty UTXOs such as the scriptpubkeys to avoid witness malleability issues ? Very unlikely we can in today tooling, what’s the UTXO sanitizing goal for the v0.1 of dual-funding support ?